scripts/eurostat/health_determinants/common/EuroStat.py

```diff
@@ -0,0 +1,359 @@
+            id_vars = ['SV', 'time']
+            var_name = 'geo'
+        if 'geo' in split_columns_list:
+            split_columns_list.remove('geo')
+            id_vars = ['SV', 'geo']
+            var_name = 'time'
+
+        df.drop(columns=split_columns_list, inplace=True)
+        df = df.melt(id_vars=id_vars,
+                     var_name=var_name,
+                     value_name='observation')
+        df = df[df['geo'] != 'EU27_2020']
+        df = df[df['geo'] != 'EU28']
+        return df
+
+    def process(self) -> list:
+        """
+        This Method calls the required methods to generate
+        cleaned CSV, MCF, and TMCF file.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        final_df = pd.DataFrame(
+            columns=['time', 'geo', 'SV', 'observation', 'Measurement_Method'])
+        # Creating Output Directory
+        output_path = os.path.dirname(self._cleaned_csv_file_path)
+        if not os.path.exists(output_path):
+            os.mkdir(output_path)
+        sv_list = []
+
+        for file_path in self._input_files:
+            df = pd.read_csv(file_path, sep='\t', header=0)
+            file_name = file_path.split("/")[-1][:-4]
+            df.columns = df.columns.str.strip()
+            df = self._parse_file(file_name, df, self._import_name)
+            df['SV'] = df['SV'].str.replace('_Total', '')
+            final_df = pd.concat([final_df, df])
+            sv_list += df["SV"].to_list()
+
+        final_df = final_df.sort_values(by=['time', 'geo', 'SV', 'observation'])
+        final_df = final_df.drop_duplicates(subset=['time', 'geo', 'SV'],
+                                            keep='first')
+        final_df['observation'] = final_df['observation'].astype(
+            str).str.strip()
+        # derived_df generated to get the year/SV/location sets
+        # where 'u' exist
+        derived_df = final_df[final_df['observation'].astype(str).str.contains(
+            'u')]
+        u_rows = list(derived_df['SV'] + derived_df['geo'])
+        final_df['info'] = final_df['SV'] + final_df['geo']
+        # Adding Measurement Method based on a condition, whereever u is found
+        # in an observation. The corresponding measurement method for all the
+        # years of that perticular SV/Country is made as Low Reliability.
+        # Eg: 2014
+        #   country/BEL
+        #   Percent_AlcoholConsumption_Daily_In_Count_Person
+        #   14.2 u,
+        # so measurement method for all 2008, 2014 and 2019 years shall be made
+        # low reliability.
+        final_df['Measurement_Method'] = np.where(
+            final_df['info'].isin(u_rows),
+            'EurostatRegionalStatistics_LowReliability',
+            'EurostatRegionalStatistics')
+        derived_df = final_df[final_df['observation'].astype(str).str.contains(
+            'd')]
+        u_rows = list(derived_df['SV'] + derived_df['geo'])
```

Comment:
same as above, should't 'time' column be added as well?

Reply:
As discussed with Prashanth, we have added same measurement method for a particular statvar for a particular geo over all the year. This is the reason 'time' is not part of u_rows. Because even if for one particular year a statvar has EurostatRegionalStatistics_LowReliability as measurement method for a geo, it should come EurostatRegionalStatistics_LowReliability as measurement method for that geo across all the year. ([Screenshot](https://screenshot.googleplex.com/4ZQ7HaZZaQZxBpZ))